libr/core/disasm.c

```diff
@@ -2628,10 +2628,22 @@ static void ds_print_ptr(RDisasmState *ds, int len, int idx) {
 			f = r_flag_get_i (core->flags, p);
 			if (f) {
 				refaddr = p;
-				if (!flag_printed && (!ds->opstr || !strstr (ds->opstr, f->name))) {
-					ALIGN;
-					ds_comment (ds, true, "; %s%s", f->name, nl);
-					flag_printed = true;
+			}
+			if (!flag_printed) {
+				list = r_flag_get_list (core->flags, p);
+				r_list_foreach (list, iter, f) {
+					if (!f->name
+					    || !strncmp (f->name, "loc.", 4)
```

Comment:
move this name filter machinery to a separate function and reuse it from both places. i understand this can cleanup the disassembly.. in the expenses of removing flags from the user which may be misleading and confusing. so im not a big fan of doing this.. maybe we can add an option after some feedback. what do you think?